neptune-core/src/application/json_rpc/server/service.rs

```diff
@@ -754,4 +764,111 @@ pub mod tests {
             );
         }
     }
+
+    #[apply(shared_tokio_runtime)]
+    async fn off_node_wallets_behave_correctly() {
+        let mut rpc_server = test_rpc_server().await;
+
+        // Prepare a transaction to our wallet coming from devnet wallet.
+        let wallet = WalletEntropy::devnet_wallet();
+        let mut devnet_node =
+            mock_genesis_global_state(0, wallet, rpc_server.state.cli().clone()).await;
+
+        let rpc_address = rpc_server
+            .state
+            .api()
+            .wallet()
+            .next_receiving_address(KeyType::Generation)
+            .await
+            .unwrap();
+        let mock_amount = NativeCurrencyAmount::coins_from_str("1").unwrap();
+        let artifacts = devnet_node
+            .api_mut()
+            .tx_sender_mut()
+            .send(
+                vec![OutputFormat::AddressAndAmount(rpc_address, mock_amount)],
+                Default::default(),
+                mock_amount,
+                Timestamp::now(),
+            )
+            .await
+            .unwrap();
+
+        // Pass transaction into rpc_server network.
+        let devnet_mempool = &devnet_node.lock_guard().await.mempool;
+        let transaction = devnet_mempool.get(artifacts.transaction().txid()).unwrap();
+        let tip =
```

Comment:
Easier if you call this `block_1` IMO. That communicates its height which makes the test a bit easier to read.